File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double sum=0;
        for(int i=0;i<k;i++)
        {
            sum=sum+nums[i];
        }
        double result=sum/k;
        for(int i=k;i<nums.length;i++)
        {
            sum=sum-nums[i-k]+nums[i];
            double avg=sum/k;
            if(result<avg)
            {
                result=avg;
            }
        }
        return result;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        add=sum(nums[:k])
        result=add/k
        for i in range(k,len(nums)):
            print(result)
            add-=nums[i-k]
            add+=nums[i]
            result=max(result,add/k)
        return result